### PR TITLE
Add support to order abstract methods

### PR DESCRIPTION
**Pull request by DarkaOnLine**
This PR allows ordering abstract methods for `order_class_elements` fixer.

New configuration keys: 
- `method_public_abstract`
- `method_public_abstract_static`
- `method_protected_abstract`
- `method_protected_abstract_static`

### CODE REVIEW

**Comment by SpacePossum** [comment]
Hi and thanks for your PR,

This looks good to me, maybe a @gharlan can have a look as well as the original author of the rule?

File: src/Fixer/ClassNotation/OrderedClassElementsFixer.php [hunk 1]
```diff
@@ -63,6 +63,10 @@ final class OrderedClassElementsFixer extends AbstractFixer implements Configura
         'method_public_static' => ['method_static', 'method_public'],
         'method_protected_static' => ['method_static', 'method_protected'],
         'method_private_static' => ['method_static', 'method_private'],
+        'method_public_abstract' => ['method_abstract', 'method_public'],
+        'method_protected_abstract' => ['method_abstract', 'method_protected'],
+        'method_public_abstract_static' => ['method_abstract_static', 'method_abstract', 'method_public'],
+        'method_protected_abstract_static' => ['method_abstract_static', 'method_abstract', 'method_protected'],
```
Comment (kubawerlos) [inline src/Fixer/ClassNotation/OrderedClassElementsFixer.php]: Shouldn't this be `method_abstract_protected_static` to follow PSR12?

**Comment by keradus** [comment]
can you help with the comments, @DarkaOnLine ? thanks!

File: src/Fixer/ClassNotation/OrderedClassElementsFixer.php (hunk 1 above)
Comment (DarkaOnLine) [reply src/Fixer/ClassNotation/OrderedClassElementsFixer.php]: changed

**Comment by DarkaOnLine** [comment]
> Can we have a test case to sort this class:
> 
> ```
> abstract class Foo
> {
>     abstract protected static function f1();
>     abstract static protected function f2();
>     protected abstract static function f3();
>     protected static abstract function f4();
>     static abstract protected function f5();
>     static protected abstract function f6();
> }
> ```
> 
> or do we assume `visibility_required` will handle it before? If latter we need to add integration test.

@kubawerlos - For now, this will not be changed. All methods are the same `abstract|protected|static` so I don't understand what should we test here. Can you please tell me that you expect here?

**Comment by DarkaOnLine** [comment]
@kubawerlos gotcha. Added additional test. Thanks for clarifying ;)

**Comment by SpacePossum** [comment]
I'm closing this one in favor of https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/4942

**Pull request closed without merging**